modules/nf-core/deepvariant/makeexamples/tests/main.nf.test

```diff
@@ -0,0 +1,189 @@
+nextflow_process {
+
+    name "Test Process DEEPVARIANT_MAKEEXAMPLES"
+    script "../main.nf"
+    config "./nextflow.config"
+    process "DEEPVARIANT_MAKEEXAMPLES"
+
+    tag "deepvariant/makeexamples"
+    tag "deepvariant"
+    tag "modules"
+    tag "modules_nfcore"
+
+    test("homo_sapiens - [bam, bai] - fasta - fai") {
+        when {
+
+            process {
+                """
+                input[0] = [
+                    [ id:'test', single_end:false ], // meta map
+                      file(params.modules_testdata_base_path + '/genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
+                      file(params.modules_testdata_base_path + '/genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam.bai', checkIfExists: true),
+                      []
+                ]
+                input[1] = [
+                    [ id:'genome'],
+                      file(params.modules_testdata_base_path + '/genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)
+                ]
+                input[2] = [
+                    [ id:'genome'],
+                      file(params.modules_testdata_base_path + '/genomics/homo_sapiens/genome/genome.fasta.fai', checkIfExists: true)
+                ]
+                input[3] = [
+                    [],[]
+                ]
+                input[4] = [
+                    [],[]
+                ]
+                """
+            }
+        }
+
+        then {
+            assertAll(
+                { assert process.success },
+                // Test string values and versions. The make_examples_id and tfrecords are not stable.
+                { assert process.out.examples.get(0).get(0) == [ id:'test', single_end:false ] },
```

Comment:
Yes this would be an improvement. The current test already assumes that there are two cpus in the process. I hope this is okay. @fellen31 do you know if there's a way to snapshot it with nf-test? Otherwise I can of course give a list of the correct names in the test code.

Reply:
The tests would fail then if someone else runs with another number of cpus, let's keep it the way you wrote it.

Reply:
The tests I have include things like

                { assert process.out.examples.get(0).get(1).size() == 4 },

So it's checking for 4 files, which is based on 2 cpus. So it does need to change. I have to see if I can reference task.cpus in the "then" code, then I can compute the correct list of file names.

Reply:
Hm, maybe you could just check if `process.out.examples` has some files, but not snapshot them... not sure.

Reply:
Can I force the number of cpus in the config?

Reply:
The tests should always work with the GitHub runners I guess, it's just if someone would test module locally. But I don't think that's very likely. It's not perfect, but maybe we can just keep it like it is.

Reply:
Right, I agree, but I can change it to check for a list of filenames. There is no downside, since both tests make an assumption of the number of cpus.

Reply:
Sounds good! Think something like this might work, since you have two files: 

```
{ assert snapshot(
    file(process.out.bam.get(0).get(1)).name,
    file(process.out.bam.get(1).get(1)).name,
).match() }
```

Reply:
Fixed now -cheking the file names